### ISSUE TITLE
Latest rock uninstallable, as there is no luke

### ISSUE THREAD

**Issue by rrthomas** [gvvaughan/lyaml#22]
build-aux/luke:628: module 'luke' not found:No LuaRocks module found for luke

**Comment by gvvaughan** [comment]
Works for me with a fresh install of lua and luarock, then `luarocks install lyaml`.

Do you have any more details of the failure? Can you build it from source (with luarocks make)? Does `build-aux/luke —debug` shoe any useful diagnostics?

**Comment by rrthomas** [comment]
There is no luke rock on luarocks.org!

-- 
https://rrt.sc3d.org

**Comment by gvvaughan** [comment]
There is no luke rock at all. It is a single minified file copied into build-aux and distributed inside the package just like a configure script.

**Comment by gvvaughan** [comment]
I mean, I don’t understand why it is trying to look for a luke rock. There’s no dependency, and it makes no sense to download the thing to regenerate the minified source at build time... and I’m unable to provoke the error message you posted :-(

**Comment by rrthomas** [comment]
Didn't realise it was so weird. Will get full trace when back at desk.

-- 
https://rrt.sc3d.org

**Comment by gvvaughan** [comment]
Lol. Okay, thanks. If you need to read the code, luke is in my github, and you can replace the minified script with a shim that invokes the readable source for intelligible line numbers and symbol names :-)

**Comment by rrthomas** [comment]
$ luarocks install lyaml
    Installing https://luarocks.org/lyaml-6.1.3-1.src.rock
    /home/rrt/.local/bin/x86_64-linux-gnu/lua5.3 build-aux/luke package="lyaml" version="6.1.3-1" PREFIX="/home/rrt/.local/lib/luarocks/rocks-5.3/lyaml/6.1.3-1" CFLAGS="-O2 -fPIC" LIBFLAG="-shared" LIB_EXTENSION="so" OBJ_EXTENSION="o" LUA="/home/rrt/.local/bin/x86_64-linux-gnu/lua5.3" LUA_DIR="/home/rrt/.local/lib/luarocks/rocks-5.3/lyaml/6.1.3-1/lua" LUA_INCDIR="/home/rrt/.local/include/lua5.3" YAML_DIR="/usr" YAML_INCDIR="/usr/include" YAML_LIBDIR="/usr/lib/x86_64-linux-gnu"
    /home/rrt/.local/bin/x86_64-linux-gnu/lua5.3: build-aux/luke:628: module 'luke' not found:No LuaRocks module found for luke
    stack traceback:
    	[C]: in function 'require'
    	build-aux/luke:628: in main chunk
    	[C]: in ?
    
    Error: Build error: Failed building.

**Comment by rrthomas** [comment]
OK, the fix is to unset my `LUA_INIT`, which says:

    LUA_INIT=@/home/rrt/.config/luarc

In turn, that file just says:

    require "luarocks.loader"

**Comment by gvvaughan** [comment]
Hmm. Maybe the minification process is interfering with that? It preloads all the modules l and relies on require fetching from the packages.preload before the other loaders. Does your LUA_INIT subvert that assumption?

It’s possibly a moot point though, because luke is a proof of concept for adding configure like host check facilities to luarocks itself - I just need to find time to reimplement the reader code for the enhanced syntax.  That done, it’ll all go away and luarocks will do all the work without a separate luke invocation.

**Comment by rrthomas** [comment]
Well, my `LUA_INIT` means (no more or less than) that all `require` invocations are going to go through luarocks's own require code, so…

**Comment by 4mig4** [comment]
I have the same problem on FreeBSD 12. 

```
luarocks install luaposix

Installing https://luarocks.org/luaposix-34.0.4-1.src.rock

/usr/local/bin/lua51 build-aux/luke package="luaposix" version="34.0.4" PREFIX="/usr/local/lib/luarocks/rocks-5.1/luaposix/34.0.4-1" LUA="/usr/local/bin/lua51" LUA_INCDIR="/usr/local/include/lua51" CFLAGS="-O2 -fPIC" LIBFLAG="-shared" LIB_EXTENSION="so" OBJ_EXTENSION="o" INST_LIBDIR="/usr/local/lib/luarocks/rocks-5.1/luaposix/34.0.4-1/lib" INST_LUADIR="/usr/local/lib/luarocks/rocks-5.1/luaposix/34.0.4-1/lua"
/usr/local/bin/lua51: build-aux/luke:658: module 'luke' not found:No LuaRocks module found for luke
stack traceback:
	[C]: in function 'require'
	build-aux/luke:658: in main chunk
	[C]: ?

Error: Failed installing dependency: https://luarocks.org/luaposix-34.0.4-1.src.rock - Build error: Failed building.
```

**Comment by gvvaughan** [comment]
Finally,  I've fixed this with release 6.2.4 by simply writing my own require implementation inside luke that only knows how to get modules from `package.preload`.

**Comment by rrthomas** [comment]
Great, thanks!
